### DISCUSSION TITLE
DockerComposeEnvironment thrown Error: (HTTP code 409) container stopped/paused in CircleCI

### DISCUSSION THREAD

**Discussion by davbr-freetrade** [testcontainers/testcontainers-node#576] [Q&A] [answered]
Hi,

I have being implementing the infrastructure to run integration test with testcontainer-node for a Typescript project.

The setp-up works fine locally. I just call the function `DockerComposeEnvironment` with the docker-compose file and it builds the services and I can both ping the API and read from the DB 👌🏻.

However, I got stuck in CircleCI trying to run the function `DockerComposeEnvironment` as it throws this error: 

```
Error: (HTTP code 409) container stopped/paused - Container 1fed6c2f4fc04c7877ceed1b7813ce783d853e69bfb273ccdb134f9c0f9ce769 is not running 
          at /home/circleci/project/node_modules/docker-modem/lib/modem.js:343:17
          at getCause (/home/circleci/project/node_modules/docker-modem/lib/modem.js:373:7)
          at Modem.Object.<anonymous>.Modem.buildPayload (/home/circleci/project/node_modules/docker-modem/lib/modem.js:342:5)
          at IncomingMessage.<anonymous> (/home/circleci/project/node_modules/docker-modem/lib/modem.js:310:16)
          at IncomingMessage.emit (node:events:525:35)
          at IncomingMessage.emit (node:domain:482:12)
          at endReadableNT (node:internal/streams/readable:1359:12)
          at processTicksAndRejections (node:internal/process/task_queues:82:21) {
        reason: 'container stopped/paused',
        statusCode: 409,
        json: {
          message: 'Container 1fed6c2f4fc04c7877ceed1b7813ce783d853e69bfb273ccdb134f9c0f9ce769 is not running'
        }
      }
```

I believe that it may have something to do with the CircleCI executor and the docker compose version so I try to with different version of `cimg/node:18.13.0`

These are the docker compose versions (V1/V2) in my local machine where everything work fine:

```
docker-compose -v
docker-compose version 1.29.2, build 5becea4c
docker compose version
Docker Compose version v2.3.3
```

These is the CircleCI image executor:

```
CircleCI image: cimg/node:18.13
build-essential 12.9ubuntu3, curl 7.81.0, docker 20.10.21, docker-compose Docker Compose version v2.12.2 v2.12.2, dockerize v0.8.0, git 2.38.1, jq 1.6, node 18.13.0, npm 8.19.3, ubuntu 22.04.1 LTS, wget 1.21.2, yarn 1.22.18
```

Have anyone in the forum try to run the DockerComposeEnvironment in CircleCI? Could it be that there is compatibilities issues between the docker compose version?

**Comment by cristianrgreco** [comment]
Hi @davbr-freetrade, could you please re-run the scenario with the environment variable `DEBUG=testcontainers*`? That will show both testcontainer logs and logs from the containers. That should help show _why_ the container is stopped.

**Reply by davbr-freetrade** [reply]
Thanks for the tips about the` DEBUG`, it is really useful! 

The initial issue was with the Postgres image, it could not create the table in the init.sql because of a path file in CircleCI

Instead of using the `.sql` file, I just pass the env variable POSTGRES_DB: <name> 

However, i updated docker compose to `version v2.17.3` and the local set-up does not work at all.

These is the error:

`Error: Log stream ended and message "/.+ Started!/" was not received`

There is not changes in the code, just docker update. The `DockerComposeEnvironment`function errors out and I have tried in two machines but it works in CircleCI with v2.12.2 🤷🏻

Have you tried this version v2.17.3?

**Reply by cristianrgreco** [reply]
> Error: Log stream ended and message "/.+ Started!/" was not received

The error here I think is quite explanative, the container is not logging as expected. If you provide the full output of `DEBUG=testcontainers*`, we'll be able to see what is being logged. It's also worth posting info like the version of Docker/Testcontainers you're using.

> Have you tried this version v2.17.3?

Yes the pipelines run both versions 1.x and 2.x of docker-compose.

**Reply by davbr-freetrade** [reply, marked as answer]
The version docker compose 2.17.3 seem to have compatible issue with "testcontainers": "^9.5.0",
I have bump to "testcontainers": "^9.8.0" and it works locally.
